product_gtin/tests/test_product_gtin_functions.py

```diff
@@ -0,0 +1,159 @@
+
+    def test_ean8_codes(self):
+        """Ean8 codes should not be valid for UPC."""
+        for code in VALID_EAN8_CODES:
+            self.assertFalse(self.barcode_mixin._check_upc(code))
+
+    def test_ean13_codes(self):
+        """Ean13 codes should not be valid for UPC."""
+        for code in VALID_EAN13_CODES:
+            self.assertFalse(self.barcode_mixin._check_upc(code))
+
+    def test_returns_earn8_codes(self):
+        for code in VALID_EAN8_CODES:
+            self.assertTrue(self.barcode_mixin._check_ean8(code))
+
+    def test_returns_wrong_ean8_codes(self):
+        self.assertFalse(self.barcode_mixin._check_ean8(""))
+        # test string
+        self.assertFalse(self.barcode_mixin._check_ean8("odoo_oca"))
+        # less than 8 numbers
+        self.assertFalse(self.barcode_mixin._check_ean8("1234567"))
+        # 8 random numbers
+        self.assertFalse(self.barcode_mixin._check_ean8("12345678"))
+        self.assertFalse(self.barcode_mixin._check_ean8("82766678"))
+        # 9 numbers
+        self.assertFalse(self.barcode_mixin._check_ean8("123456789"))
+
+    def test_return_ean8_codes(self):
+        """Ean8 should not accept ean13"""
+        for code in VALID_EAN13_CODES:
+            self.assertFalse(self.barcode_mixin._check_ean8(code))
+
+    def test_return_upc_codes(self):
+        """Ean8 should not accept UPC"""
+        for code in VALID_UPC_CODES:
+            self.assertFalse(self.barcode_mixin._check_ean8(code))
+
+    def test_return_ean13_codes(self):
+        """test valid ean 13 number."""
+        for code in VALID_EAN13_CODES:
+            self.assertTrue(self.barcode_mixin._check_ean13(code))
+
+    def test_wrong_ean13_codes(self):
+        self.assertFalse(self.barcode_mixin._check_ean13(""))
+        # test string
+        self.assertFalse(self.barcode_mixin._check_ean8("odoo_oca_sflx"))
+        # less than 13 numbers
+        self.assertFalse(self.barcode_mixin._check_ean13("123456789012"))
+        # 13 random numbers
+        self.assertFalse(self.barcode_mixin._check_ean13("1234567890123"))
+        self.assertFalse(self.barcode_mixin._check_ean13("1234514728123"))
+        # 14 numbers
+        self.assertFalse(self.barcode_mixin._check_ean13("12345147281234"))
+
+    def test_returns_ean8_codes(self):
+        """Ean13 should not accept ean8"""
+        for code in VALID_EAN8_CODES:
+            self.assertFalse(self.barcode_mixin._check_ean13(code))
+
+    def test_returns_upc_codes(self):
+        """Ean13 should not accept UPC"""
+        for code in VALID_UPC_CODES:
+            self.assertFalse(self.barcode_mixin._check_ean13(code))
+
+    def test_DICT_CHECK_CODE_TYPE(self):
+        """Check if the dict _DICT_CHECK_CODE_TYPE exists."""
+        self.assertTrue(self.barcode_mixin._DICT_CHECK_CODE_TYPE)
+
+    def test_DICT_CHECK_FUNCTIONS(self):
+        """Check if the dict _DICT_CHECK_FUNCTIONS exists."""
+        self.assertTrue(self.barcode_mixin._DICT_CHECK_FUNCTIONS)
```

Comment:
You could probably concentrate some of these tests in a singe one, without the need for a function name so specific...
```suggestion
    def test_dicts_check_exist(self):
        """Check if the dicts _DICT_CHECK_CODE_TYPE and _DICT_CHECK_FUNCTIONS exist."""
        self.assertTrue(self.barcode_mixin._DICT_CHECK_CODE_TYPE)
        self.assertTrue(self.barcode_mixin._DICT_CHECK_FUNCTIONS)
```

The same applies to some of the other functions that just have 1 or 2 lines, if you think it is appropriate.

Reply:
Thanks, i have improved some functions to simplify validations.